indico/modules/categories/forms.py

```diff
@@ -119,57 +142,103 @@ def __init__(self, *args, category, **kwargs):
-                    _('The credentials are invalid or have no access to the Google Wallet API.')
-                )
-                return False
-            elif res == GoogleCredentialValidationResult.failed:
-                self.google_wallet_credentials.errors.append(
-                    _('There was an error validating your credentials. Contact your Indico administrator for details.')
-                )
-                return False
-            elif res == GoogleCredentialValidationResult.bad_issuer:
-                self.google_wallet_issuer_id.errors.append(
-                    _('The Issuer ID is invalid or not linked to your credentials.')
-                )
-                return False
+            ):
+                res = GoogleWalletManager.verify_credentials(credentials, issuer_id)
+                if res == GoogleCredentialValidationResult.invalid:
+                    self.google_wallet_credentials.errors.append(
+                        _('The credentials could not be loaded.')
+                    )
+                    return False
+                elif res == GoogleCredentialValidationResult.refused:
+                    self.google_wallet_credentials.errors.append(
+                        _('The credentials are invalid or have no access to the Google Wallet API.')
+                    )
+                    return False
+                elif res == GoogleCredentialValidationResult.failed:
+                    self.google_wallet_credentials.errors.append(
+                        _('There was an error validating your credentials. Contact your Indico administrator for '
+                          'details.')
+                    )
+                    return False
+                elif res == GoogleCredentialValidationResult.bad_issuer:
+                    self.google_wallet_issuer_id.errors.append(
+                        _('The Issuer ID is invalid or not linked to your credentials.')
+                    )
+                    return False
 
         return form_valid
 
+    def validate_apple_pass_certificate(self, field):
+        try:
+            x509.load_pem_x509_certificate(field.data.encode('UTF-8'))
+            return True
+        except ValueError:
+            raise ValidationError(_('The provided Certificate.pem is malformed.'))
+
+    def validate_apple_pass_key(self, field):
+        try:
+            serialization.load_pem_private_key(field.data.encode('UTF-8'), password=None)
+        except ValueError:
+            raise ValidationError(_('The provided Private.key is malformed.'))
+        except TypeError:
+            # This is because I provided an empty pass, just to check the Private.key is not malformed.
+            # Is there a better way to check it?
+            return True
+        except UnsupportedAlgorithm:
+            raise ValidationError(_('The provided Private.key type is not supported by the system. '
+                                    'Please contact support.'))
```

Comment:
maybe failing with a generic "the private key is invalid" message is enough (and just log something more detailed)

Reply:
I trapped the UnsupportedAlgorithm because of the hazmat official documentation:

https://cryptography.io/en/latest/hazmat/primitives/asymmetric/serialization/

@OmeGak so what should I do?

Reply:
I'm not questioning catching `UnsupportedAlgorithm` but the error message that will be displayed to the user. @ThiefMaster's suggestion for a more generic "the private key is invalid" error message seems reasonable to me.

Reply:
Done.